snippets/scripts/compass-github-importer/index.js

```diff
@@ -0,0 +1,174 @@
+        method: 'POST',
+        headers: {
+            Authorization: `Basic ${header}`,
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(query),
+    }).then((res) => res.json());
+}
+
+// This function looks for a component with a certain name, but if it can't find it, it will create a component.
+async function putComponent(componentName, description, web_url) {
+    const response = await makeGqlRequest({
+        query: `
+      query getComponent {
+        compass @optIn(to: "compass-beta") {
+          searchComponents(cloudId: "${CLOUD_ID}", query: {
+            query: "${componentName}",
+            first: 1
+          }) {
+            ... on CompassSearchComponentConnection {
+              nodes {
+                component {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+      `,
+    });
+    const maybeResults = response?.data?.compass?.searchComponents?.nodes;
+
+    if (!Array.isArray(maybeResults)) {
+        console.error(`error fetching component: `, JSON.stringify(response));
+        throw new Error('Error fetching component');
+    }
+
+    const maybeComponentAri = maybeResults.find(
+        (r) => r.component?.name === componentName
+    )?.component?.id;
+    if (maybeComponentAri) {
+        console.log(chalk.gray(`Already added ${web_url} ... skipping`))
+        return maybeComponentAri;
+    } else {
+        if (!dryRun) {
+            const response = await makeGqlRequest({
+                query: `
+        mutation createComponent {
+          compass @optIn(to: "compass-beta") {
+            createComponent(cloudId: "${CLOUD_ID}", input: {name: "${componentName}", description: "${description}" typeId: "OTHER", links: [{type: REPOSITORY, name: "Repository", url: "${web_url}"}]}) {
+              success
+              componentDetails {
+                id
+              }
+            }
+          }
+        }`,
+            });
+
+            const maybeAri =
+                response?.data.compass?.createComponent?.componentDetails?.id;
+            const isSuccess = !!response?.data.compass?.createComponent?.success;
+            if (!isSuccess || !maybeAri) {
+                console.error(`error creating component: `, JSON.stringify(response));
+                throw new Error('Could not create component');
+            }
+            console.log(chalk.green(`New component for ${web_url} ... added ${componentName}`))
+            return maybeAri;
+        } else {
+            console.log(chalk.yellow(`New component for ${web_url} ... would be added ${componentName} (dry-run)`))
+            return;
+        }
+    }
+}
+
+async function listAllOrgs() {
+    let instanceOrganizations = []
+    let page = 1;
+    const perPage = 100; // Maximum items per page as allowed by GitHub Enterprise Server API
+
+    while(true) {
```

Comment:
Is there any way we can replace this with a for loop based on `page`, or avoid the "while true"?